src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -133,29 +129,32 @@ export class EdgeProvider extends Bridgeable {
   }
 
   // Get an address from the user's wallet
-  getReceiveAddress(options: EdgeGetReceiveAddressOptions): EdgeReceiveAddress {
-    const wallet: GuiWallet = this._state.ui.wallets.byId[this._state.ui.wallets.selectedWalletId]
+  async getReceiveAddress(options: EdgeGetReceiveAddressOptions): Promise<EdgeReceiveAddress> {
+    const coreWallet: EdgeCurrencyWallet = this._state.core.account.currencyWallets[this._state.ui.wallets.selectedWalletId]
+    const receiveAddress = await coreWallet.getReceiveAddress()
     if (options && options.metadata) {
-      wallet.receiveAddress.metadata = options.metadata
+      receiveAddress.metadata = options.metadata
     }
-    return Promise.resolve(wallet.receiveAddress)
+    return receiveAddress
   }
 
-  getCurrentWalletInfo(): Promise<WalletDetails> {
-    const wallet: GuiWallet = this._state.ui.wallets.byId[this._state.ui.wallets.selectedWalletId]
-    const currentCode = this._state.ui.wallets.selectedCurrencyCode
-    let walletName = wallet.name
-    if (wallet.enabledTokens.length > 1) {
+  async getCurrentWalletInfo(): Promise<WalletDetails> {
+    const coreWallet = this._state.core.account.currencyWallets[this._state.ui.wallets.selectedWalletId]
+    const currencyCode = this._state.ui.wallets.selectedCurrencyCode
+    let walletName = coreWallet.name ? coreWallet.name : ''
+    const receiveAddress = await coreWallet.getReceiveAddress()
+    const enabledTokens = await coreWallet.getEnabledTokens()
+    if (enabledTokens.length > 1) {
       console.log('EP: We have tokens.. what do we do with them ')
-      walletName = currentCode
+      walletName = currencyCode
     }
     const returnObject: WalletDetails = {
       name: walletName,
-      receiveAddress: wallet.receiveAddress,
-      currencyCode: currentCode,
-      fiatCurrencyCode: wallet.fiatCurrencyCode,
-      currencyIcon: wallet.symbolImage,
-      currencyIconDark: wallet.symbolImageDarkMono
+      receiveAddress,
+      currencyCode,
+      fiatCurrencyCode: isoFiatSplatter(coreWallet.fiatCurrencyCode).fiatCurrencyCode,
+      currencyIcon: coreWallet.currencyInfo.symbolImage, // TODO: get images for tokens
```

Comment:
We have a function you can use to coverts a pluginId + contractAddress (if it's a token) to the correct icon links: https://github.com/EdgeApp/edge-react-gui/blob/64c512fe17bc196aa8b1e9aed9215a719967edd3/src/util/CurrencyInfoHelpers.js#L21

Reply:
Single English word variable names such as wallet are very ambiguous. I am open to edgeWallet which I am just now noticing is used in other places.

Reply:
I was only trying to remove the use of GUI Wallet and not change additional functionality like how we get icons

Reply:
That deserves a different PR